### PR TITLE
Refactoring of format and mount in expert partitioner

### PR DESCRIPTION
**Pull request by ancorgs**
- Refactored the add_partition and edit_blk_device workflows to work directly in y2storage objects (which enables us to work in two PBIs).
- Centralized the logic in new controller classes.
- Meaningful unit tests for the controllers
- Added button to manage btrfs subvolumes directly during creation/modification
- Fixed several small bugs about forms behavior, detection of preconditions and other topics:
  - https://bugzilla.suse.com/show_bug.cgi?id=1057869
  - https://bugzilla.suse.com/show_bug.cgi?id=1057874
  - Others (some not reported yet)

### CODE REVIEW

**Comment by ancorgs** [comment]
@teclator You are probably the best candidate to review this. I guess it's better understood commit by commit

File: src/lib/y2partitioner/dialogs/btrfs_subvolume.rb [hunk 1]
```diff
@@ -97,18 +97,14 @@ def init
 
         # Validates the subvolume path
         #
-        # @note The subvolume shadowing is also checked due to its mount point
-        #   is generated from the subvolume path.
-        #
         # The following condintions are checked:
```
Comment (teclator) [inline src/lib/y2partitioner/dialogs/btrfs_subvolume.rb]: typo: conditions

File: src/lib/y2partitioner/dialogs/btrfs_subvolume.rb [hunk 2]
```diff
@@ -97,18 +97,14 @@ def init
...
         # - The subvolume path is not empty
         # - The subvolume path starts by the default subvolume path
         # - The subvolume path in unique for the filesystem
```
Comment (teclator) [inline src/lib/y2partitioner/dialogs/btrfs_subvolume.rb]: typo: is unique

File: src/lib/y2partitioner/sequences/filesystem_controller.rb [hunk 3]
```diff
@@ -0,0 +1,316 @@
...
+
+      def mount_point
+        filesystem ? filesystem.mountpoint : nil
+      end
+
+      def partition_id
+        blk_device.is?(:partition) ? blk_device.id : nil
+      end
+
+      def apply_role
```
Comment (teclator) [inline src/lib/y2partitioner/sequences/filesystem_controller.rb]: NP: In general miss more methods documentation in this class but in the rest of classes it is well documented

Comment (ancorgs) [reply src/lib/y2partitioner/sequences/filesystem_controller.rb]: Oops, this was overlooked. Please check the new commit including your fixes and the documentation for this class.

Comment (teclator) [reply src/lib/y2partitioner/sequences/filesystem_controller.rb]: Thnx!!!

**Comment by teclator** [comment]
And now it looks even better 👍

**Pull request merged**